examples/medplum-task-example-app/app/src/components/task-actions/AddDueDate.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button, Modal } from '@mantine/core';
+import { notifications } from '@mantine/notifications';
+import { getQuestionnaireAnswers, MedplumClient, normalizeErrorString, PatchOperation } from '@medplum/core';
+import { Period, Questionnaire, QuestionnaireResponse, Task, TaskRestriction } from '@medplum/fhirtypes';
+import { QuestionnaireForm, useMedplum } from '@medplum/react';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { useState } from 'react';
+
+interface AddDueDateProps {
+  task: Task;
+  onChange: (updatedTask: Task) => void;
+  medplum: MedplumClient;
+}
+
+export function AddDueDate(props: AddDueDateProps): JSX.Element {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const handleOpenClose = (): void => {
+    setIsOpen(!isOpen);
+  };
+
+  const handleAddDueDate = async (
+    date: string,
+    task: Task,
+    medplum: MedplumClient,
+    onChange: (task: Task) => void
+  ): Promise<void> => {
+    const ops: PatchOperation[] = [{ op: 'test', path: '/meta/versionId', value: task.meta?.versionId }];
+
+    // Create a restriction period with the due-date if one doesn't already exist and add it to the PatchOperation. For more details on task due-dates, see https://www.medplum.com/docs/careplans/tasks#task-start--due-dates
+    if (!task.restriction) {
```

Comment:
You can refactor this a bit: 

- Make a variable called 'op' that is either set to 'add' or 'replace'
- define the `restriction` outside of the `if/else`, and set accordingly